### ISSUE TITLE
Support metalava --proguard option

### ISSUE THREAD

**Issue by herriojr** [tylerbwong/metalava-gradle#40]
There is an option for metalava to generate a proguard keep file. This should be exposed.

**Comment by tylerbwong** [comment]
Good idea! Curious to know your thoughts on how it should be exposed. Would you rather have it as a separate Gradle task (e.g. `metalavaGenerateKeepFile`) with the same inputs as `metalavaGenerateSignature`? Or would you rather have it as an option that is configurable through the `MetalavaExtension` as
```
metalava {
    filename.set("api/api.txt")
    generateKeepFiles.set(true)
    keepFilename.set("proguard-rules.txt") // Optionally configure keep file name
}
```

**Comment by herriojr** [comment]
@tylerbwong I think that's somewhat of a question on whether it has to run through the signature generation again or not. Meaning, is it going to double the execution time if we separate them or will it only consist of the added overhead of the keepfile generation. If it's going to double the time to separate them, then I'd just have it generated as part of the signature generation. I was looking into this the other day but got side tracked. I added it as part of the signature, and I got far enough along to generate the proguard rules but I also would receive a non-zero exit status with no explanation of what that I needed to look into further.
